request-manager/graph/grapher.go

```diff
@@ -217,25 +236,27 @@ func buildSeqGraph(seqSpec *spec.Sequence, idgen id.Generator) (seqGraph *Graph,
 		}
 	}
 
-	// Nodes we've yet to add
+	// Nodes we've yet to add, represented as single-node graphs
 	nodesToAdd := map[string]*Graph{}
 	for k, v := range nodes {
 		nodesToAdd[k] = v
```

Comment:
You can do this in the previous loop: i.e. create the nodes and add them to `nodesToAdd`. It's an easy micro-optimization.